open-apparel-registry/open-apparel-registry
# Update ESLint rules, add lint script, and fix or circumvent project lint

## Overview

This PR adjusts the React app's linting rules to match our usual project set up, and also adds a lint script which is now run during cibuild.

I made a couple of passes to fix project lint:

First, I renamed JSX files to have JSX extensions when possible, which meets a required rule that files containing JSX have that filename

Next I ran Prettier over the project using the config from ISM

Then I ran eslint --fix which remedied a bunch of trivial things

Finally I made some manual adjustments to specific files to...

1. fix some trivial issues, like making sure Link tags had both to and href props
2. make line-specific eslint-disable comments for issues which require some pinpoint fixes
3. make file-wide eslint-disable comments for specific rules which would require more involved fixes

Naming the rules like that makes it possible for us to circle back to them. In particular, there are some currently-disabled JSX-A11Y rules which it seems like we should address.

Connects #21 
Connects #22 

## Notes

I also included the commit to pin React dependencies on this branch, so it can be reviewed along with this one.

## Testing Instructions

This has some changes to eslint versions, so you'll have to reinstall your node_modules, then

- run `./scripts/lint` to verify that  you don't see any linter errors
- run `docker-compose up react` to verify that the project can still compile
